src/components/Application/BriefOwnerHeader.tsx

```diff
@@ -56,6 +68,55 @@ const BriefOwnerHeader = (props: BriefOwnerHeaderProps) => {
         setAnchorEl(null);
     };
 
+    const getBalance = useCallback(async (walletAddress: string, currency_id: number) => {
+        try {
+            const imbueApi = await initImbueAPIInfo();
+            const chainService = new ChainService(imbueApi, user);
+
+            if (!walletAddress) return "No Wallet Found"
+            const balance: any = await chainService.getBalance(walletAddress, currency_id)
+            return balance
+
+        } catch (error) {
+            setError(error)
```

Comment:
I think you should return a string indicating that an error occured while fetching the balance.

Reply:
though the catch block should create a message with the cause. but anyways I'll add a string message as you said.